### ISSUE TITLE
Grandia FMV Playback Slow

### ISSUE THREAD

**Issue by Alextended** [libretro/beetle-saturn-libretro#35]
I've tried everything, as soon as any Grandia FMV starts the framerate slows so playback is slow and the audio is distorted.

It runs fine in the latest Mednafen and the issue doesn't come up in any other game that I've noticed, in-game or in FMV...

I believe it might be related to this but I'm not entirely sure:
https://github.com/libretro/beetle-saturn-libretro/issues/24

Please fix it someone out there :)

**Comment by McRmie** [comment]
Reproduced the issue on my end with the latest beetle_saturn core. Lots of stuttering/slowdowns during the FMVs but then I wondered if it was simply a performance issue. 

Sure enough, I upped the audio latency and changed to vulkan instead of glcore (apparently vulkan can be faster than glcore even with non hardware-accelerated cores it seems) and the stutterings/slowdowns during FMV were gone. The FMVs in the game just seem to be very cpu demanding for some reason (maybe video decoding related?).

Try changing some of the settings/drivers in RA -things that might be more demanding cpu wise. Try upping the audio_latency for the game for example. Switch the video driver, or remove some of the lag-reducing options etc. Anything that might lower the cpu costs and see if that makes a difference. Remember that Saturn emulation (and beetle saturn especially) is already relatively demanding.

edit: Am on Windows -Ryzen 7 2700. 
edit: Enabling threaded video in addition might help a lot as well (though might add a bit of input lag I believe). Standalone 1.32.1 mednafen saturn might have added additional optimizations which may be why you get full speed or it's the lag-reducing options of RA mentioned above that might add to the requirements for full speed.

**Comment by Vamp898** [comment]
For reference, I tried all of this but nothing helped. FMV in other games and the game itself works fine, but not the FMV in Grandia.

No issues in mednafen

**Comment by Alextended** [comment]
Yeah, the issue has persisted for me as well throughout all this time.

Mind I recently got a new (modest, no flagship or anything) phone, the Poco X7 Pro, and it's strong enough for Beetle Saturn and the issue with these FMV does NOT occur there so it's something specific to the Windows build or however these work.

**Comment by Vamp898** [comment]
I noticed, it does work perfect on Linux too. Not on my Android phone (Maybe it is too slow? But it can play FMV Videos from other games without any issues whatsoever and it can play the actual game) but on my Linux Desktop, there is no issue with the Beetle Saturn Core

**Comment by pstef** [comment]
Should be better after https://github.com/libretro/beetle-saturn-libretro/pull/64

**Comment by Alextended** [comment]
Grandia FMV performance seems fixed! New graphical glitches after the game boots though! Big green vertical line and in game various horizontal ones too after loading asave. But thanks, good sign!

**Comment by pstef** [comment]
Will look into that, thanks for reporting.

**Comment by Alextended** [comment]
Here's some screenshots if they help;
https://i.ibb.co/1GKkSkzb/retroarch-2026-05-12-12-46-44-034.png
https://i.ibb.co/XrDBCx7L/retroarch-2026-05-12-12-40-30-776.png

BUT. While I'm on latest windows 11/latest retroarch/updated cores/everything as always and with the same settings like Vulkan etc., and these issues had never shown before and ra and most cores I use have been just fine. I should mention my GPU is end of life and no longer supported. I have a GTX1080 and I'm using the last hotfix drivers available for it.
https://www.nvidia.com/en-us/drivers/details/263265/
So maybe not something you should bother with if it's a driver issue for no longer supported GPUs only. Just saying.

 Gungriffon only gets a less obvious vertical line in FMV but fine in game. Virtua Fighter 2 seems unaffected. Haven't tried more yet.

Edit: similar glitches on my Poco X7 Pro too after updating everything there also (not play store retroarch, downloaded from site).

**Comment by LibretroAdmin** [comment]
Panzer Dragoon issue should be fixed by this
https://github.com/libretro/beetle-saturn-libretro/commit/dfb3c267689c7d3eb7a4e44f64cdeb4fb7fe8b0c
I think it fixes Grandia too, can't reproduce the issues here anymore

**Comment by Alextended** [comment]
> Panzer Dragoon issue should be fixed by this [dfb3c26](https://github.com/libretro/beetle-saturn-libretro/commit/dfb3c267689c7d3eb7a4e44f64cdeb4fb7fe8b0c) I think it fixes Grandia too, can't reproduce the issues here anymore

Every issue encountered after the initial Grandia FMV performance fix seems to be back to normal (in the games tried so far).

Whatever you did also fixed the small performance issues I had in Last Bronx and Touge King the Spirits 2 (I never reported it because I chalked it up to my ancient PC build, while it didn't make sense to me for Grandia FMV playback to be more demanding).

Thanks!!!!!!

**Comment by pstef** [comment]
I think we can close this and https://github.com/libretro/beetle-saturn-libretro/issues/24.

Possibly also https://github.com/libretro/beetle-saturn-libretro/issues/39, https://github.com/libretro/beetle-saturn-libretro/issues/50, https://github.com/libretro/beetle-saturn-libretro/issues/49

**Comment by Alextended** [comment]
It does seem to run worse on my Poco X7 Pro now mind. But don't trust me too much, I hadn't tested thoroughly but as I said in way older comment it seemed to not have the Grandia FMV issues which it now seems to have some and also frame drops in VF2 I tested. But yeah seems like these should be solved for most and Poco X7 Pro isn't a flagship phone or anything (but not bad). Just saying!

**Comment by Vamp898** [comment]
The Mediatek Dimensity 8400 Ultra should be fast enough for Saturn Emulation with Mednafen, it far exceeds it's requirements

**Comment by Alextended** [comment]
That seems fixed too now :) Edit: maybe still some drops in VF2.

**Comment by Alextended** [comment]
Guess this is still being tweaked? Earlier today the issue seemed to occur in certain 2-3 spots in the intro (nowhere near as bad as when this thread was made or before the fixes discussed in recent posts as it affected the whole video), now it's fine after updating again. Also the performance issues I mentioned on my phone after these fixes were introduced seem to be taken care of except if I switch the phone (and screen) orientation while in game, then it gets some hitching that won't be fixed until closing and running a game again, Idk if that's a core or RA bug, I haven't noticed it before, just saying.

**Comment by McRmie** [comment]
Just saw Beetle Saturn was updated. 

Regarding the issue with Grandia's FMV; this is/was less of a bug and more of a performance/optimization issue. For some reason Grandia's FMVs just seem to demand a lot of processing power from the Saturn (and naturally, will be even more demanding to emulate during those parts)

Guessing it's using some kind of (software) video decoder or something like that for playing its videos which is why the performance hiccups occurs/occurred when the video is more "busy" visually. So the Saturn would be doing more work at certain points decoding the video.

Basically, if your system is fast enough to reach 60fps with most games most of the time, it might drop below 60 in those cases. Though as I mentioned, disabling threaded video gave me enough of a performance boost that even with 1.29. I didn't get any hiccups (and without adding any noticeable input lag -worse case scenario maybe disabling threaded video adds 1 frame) 

Anyway, it seems Mednafen 1.32 has some more optimizations there so the video hiccups on Grandia should occur even more rarely on Beetle Saturn 1.32, but again, if you still experience some hiccups, disabling certain sync features like threaded video helps a lot I found.

**Comment by Alextended** [comment]
I did notice that even in Ymir emulator I was testing the Grandia FMVs ran very slowly but disabling vsync fixed it and I had tested that with Beetle just in case but it didn't work, but anyway as mentioned earlier it seems 100% fixed, at the moment, on both my ancient PC (i7 3770K/GTX1080) and my phone. I suppose for an even lesser PC it might still expose an issue other games don't.

**Comment by Vamp898** [comment]
I did not have any issues with either Mednafen 1.29 nor 1.32, only with Beetle Saturn.

**Comment by geysonazevedo27** [comment]
Some games still experience slowdown issues on the recent Beetle Saturn from RetroArch.
